tests/end_to_end/andor_assignment/orig.c

```diff
@@ -13,6 +13,20 @@ int test(int a, int b, int c, int d) {
     ret = 0;
     if (var1 || var2 || (var2 = foo(var2)) || d) {
         ret = 1;
+    } else if (a) {
+        ret = -1;
+    } else {
+        ret = -2;
     }
+    ret += c;
+
+    if (var1 && var2 && ((var1 += var2), (var2 = foo(var1))) && d) {
+        while (ret < 5) {
+            ret += 1;
+            ret *= 2;
+        }
+        ret += 5;
+    }
+
     return ret;
 }
```

Comment:
can you add another test for mixed &&/||? output won't be perfect ofc, but we need to make sure we don't crash or do something wildly wrong